users/views.py

```diff
@@ -1,3 +1,48 @@
-from django.shortcuts import render
+import json
+import re
+from json.decoder import JSONDecodeError
 
-# Create your views here.
+from django.views import View
+from django.http  import JsonResponse
+from django.db    import IntegrityError 
+
+from users.models import User
+
+class SignUpView(View):
+    def post(self,request):
+        try:
+            data = json.loads(request.body)
+
+            REGEX_EMAIL        = re.compile(r'^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$')
+            REGEX_PASSWORD     = re.compile(r'^[a-zA-Z0-9]{8,20}$')
+            REGEX_PHONE_NUMBER = re.compile(r'^01([0|1|6|7|8|9])([0-9]{3,4})([0-9]{4})$')
+            REGEX_NAME         = re.compile(r'^[가-힣]{2,5}|[a-zA-Z]{2,20}\s[a-zA-Z]{2,20}$')
+
+            if not REGEX_EMAIL.match(data["email"]):
+                return JsonResponse({"message" : "INVALID_EMAIL_FORMAT"} , status = 400) 
+            if not REGEX_PASSWORD.match(data["password"]):
+                return JsonResponse({"message" : "INVALID_PASSWORD_FORMAT"} , status = 400)
+            if not REGEX_PHONE_NUMBER.match(data["phone_number"]):
+                return JsonResponse({"message": "INVALID_EMAIL_FORMAT"} , status = 400)
+            if not REGEX_NAME.match(data["name"]):
+                return JsonResponse({"message" : "INVALID_NAME_FORMAT"} , status = 400) 
```

Comment:
```suggestion
            if not REGEX_EMAIL.match(data["email"]):
                return JsonResponse({"message" : "INVALID_EMAIL_FORMAT"} , status = 400)
                 
            if not REGEX_PASSWORD.match(data["password"]):
                return JsonResponse({"message" : "INVALID_PASSWORD_FORMAT"} , status = 400)
                
            if not REGEX_PHONE_NUMBER.match(data["phone_number"]):
                return JsonResponse({"message": "INVALID_EMAIL_FORMAT"} , status = 400)
                
            if not REGEX_NAME.match(data["name"]):
                return JsonResponse({"message" : "INVALID_NAME_FORMAT"} , status = 400) 
```
핸드폰 번호 포멧 확인하는 곳의 에러 메시지 확인 부탁드립니다.

Reply:
수정했습니다.